packages/api/src/name.js

```diff
@@ -65,5 +68,111 @@ export async function namePost (request, env) {
     new PreciseDate(uint8ArrayToString(entry.validity)).getFullTime()
   )
 
+  ctx.waitUntil((async () => {
+    const record = await env.db.resolveNameRecord(key)
+    if (!record) return // shouldn't happen
+    const { value } = ipns.unmarshal(uint8ArrayFromString(record, 'base64pad'))
+    const data = { key, value: uint8ArrayToString(value), record }
+    await NameRoom.broadcast(request, env.NAME_ROOM, key, data)
+    await NameRoom.broadcast(request, env.NAME_ROOM, '*', data)
+  })())
+
   return new JSONResponse({ id: key }, { status: 202 })
 }
+
+/**
+ * @param {Request} request
+ * @param {import('./env').Env} env
+ */
+export async function nameWatchGet (request, env) {
+  const { params: { key } } = request
+  if (key === '*') {
+    return NameRoom.join(request, env.NAME_ROOM, '*')
+  }
+
+  const keyCid = CID.parse(key, base36)
+  if (keyCid.code !== libp2pKeyCode) {
+    throw new HTTPError(`invalid key code: ${keyCid.code}`, 400)
+  }
+
+  return NameRoom.join(request, env.NAME_ROOM, key)
+}
+
+/**
+ * A Cloudflare Durable Object. Each instance is a group of users interested in
+ * a particular IPNS key ID. The static API is used by workers to communicate
+ * with it.
+ */
+export class NameRoom {
+  constructor () {
+    /** @type {Array<{ webSocket: WebSocket }>} */
```

Comment:
Any reason to be an array of Objects instead of an array of Websocket instances?

Reply:
Not really, I guess I was thinking we might want to attach some other session information in the future...